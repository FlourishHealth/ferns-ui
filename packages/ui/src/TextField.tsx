import {getCalendars} from "expo-localization";
import React, {FC, useMemo, useState} from "react";
import {
  DimensionValue,
  KeyboardTypeOptions,
  Platform,
  Pressable,
  StyleProp,
  TextInput,
  View,
} from "react-native";

import {TextFieldProps, TextStyleWithOutline} from "./Common";
import {FieldError, FieldHelperText, FieldTitle} from "./fieldElements";
import {Icon} from "./Icon";
import {useTheme} from "./Theme";

const keyboardMap: {[id: string]: string | undefined} = {
  date: "default",
  email: "email-address",
  number: "number-pad",
  numberRange: "number-pad",
  decimalRange: "decimal-pad",
  decimal: "decimal-pad",
  height: "default",
  password: "default",
  phoneNumber: "number-pad",
  search: "default",
  text: "default",
  url: Platform.select({
    ios: "url",
    android: "default",
  }),
  username: "default",
};

// Not an exhaustive list of all the textContent types, but the ones we use.
const textContentMap: {
  [id: string]: "none" | "emailAddress" | "password" | "username" | "URL" | undefined;
} = {
  date: "none",
  email: "emailAddress",
  number: "none",
  decimal: "none",
  decimalRange: "none",
  height: "none",
  password: "password",
  search: "none",
  text: "none",
  url: Platform.select({
    ios: "URL",
    android: "none",
  }),
  username: "username",
};

export const TextField: FC<TextFieldProps> = ({
  title,
  disabled,
  helperText,
  errorText,
  value,
  onChange,
  placeholder,
  blurOnSubmit = true,
  iconName,
  onIconClick,
  type = "text",
  autoComplete,
  inputRef,
  multiline,
  rows = 1,
  grow,
  returnKeyType,
  onBlur,
  onFocus,
  onEnter,
  onSubmitEditing,
  testID,
}) => {
  const {theme} = useTheme();

  const calendar = getCalendars()[0];
  const localTimeZone = calendar?.timeZone;
  if (!localTimeZone) {
    console.warn("Could not automatically determine timezone.");
  }

  const [focused, setFocused] = useState(false);
  const [height, setHeight] = useState(rows * 40);

  let borderColor = focused ? theme.border.focus : theme.border.dark;
  if (disabled) {
    borderColor = theme.border.activeNeutral;
  } else if (errorText) {
    borderColor = theme.border.error;
  }

  const calculatedHeight: DimensionValue = useMemo(() => {
    if (grow) {
      return Math.max(40, height);
    } else if (multiline) {
      return height || "100%";
    } else {
      return 20;
    }
  }, [grow, height, multiline]);

  const defaultTextInputStyles = useMemo(() => {
    const style: StyleProp<TextStyleWithOutline> = {
      flex: 1,
      width: "100%",
      height: calculatedHeight,
      color: theme.text.primary,
      fontFamily: "text",
      fontSize: 16,
      paddingVertical: 0,
      gap: 10,
    };

    if (Platform.OS === "web") {
      style.outline = "none";
    }
    return style;
  }, [calculatedHeight, theme.text.primary]);

  if (["numberRange", "decimalRange", "height"].includes(type)) {
    console.warn(`${type} is not yet supported`);
  }

  const shouldAutocorrect =
    ["text", "textarea"].includes(type) && (!autoComplete || autoComplete === "on");

  const keyboardType = keyboardMap[type];
  const textContentType = textContentMap[type || "text"];

  return (
    <View
      style={{
        flexDirection: "column",
        width: "100%",
      }}
    >
      {Boolean(title) && <FieldTitle text={title!} />}
      {Boolean(errorText) && <FieldError text={errorText!} />}
      <View
        style={{
          flexDirection: "row",
          alignItems: "center",
          backgroundColor: disabled ? theme.surface.neutralLight : theme.surface.base,
          borderColor,
          borderWidth: focused ? 3 : 1,
          paddingHorizontal: focused ? 10 : 12,
          paddingVertical: focused ? 6 : 8,
          borderRadius: 4,
          overflow: "hidden",
        }}
      >
        <TextInput
          ref={(ref) => {
            if (inputRef) {
              inputRef(ref);
            }
          }}
          accessibilityHint="Enter text here"
          accessibilityState={{disabled}}
          aria-label="Text input field"
          autoCapitalize={type === "text" ? "sentences" : "none"}
          autoCorrect={shouldAutocorrect}
          blurOnSubmit={blurOnSubmit}
          enterKeyHint={returnKeyType}
          keyboardType={keyboardType as KeyboardTypeOptions}
          multiline={multiline}
          numberOfLines={rows || 4}
          placeholder={placeholder}
          placeholderTextColor={theme.text.secondaryLight}
          readOnly={disabled}
          secureTextEntry={type === "password"}
          style={defaultTextInputStyles}
          testID={testID}
          textContentType={textContentType}
          underlineColorAndroid="transparent"
          value={value}
          onBlur={() => {
            if (disabled) return;
            if (onBlur) {
              onBlur(value ?? "");
            }
            setFocused(false);
          }}
          onChangeText={onChange}
          onContentSizeChange={(event) => {
            if (!grow) {
              return;
            }
            setHeight(event.nativeEvent.contentSize.height);
          }}
          onFocus={() => {
            if (!disabled) {
              setFocused(true);
            }
            if (onFocus) {
              onFocus();
            }
          }}
          onSubmitEditing={() => {
            if (onEnter) {
              onEnter();
            }
            if (onSubmitEditing) {
              onSubmitEditing();
            }
          }}
        />
        {Boolean(iconName) && (
          <Pressable aria-role="button" onPress={onIconClick}>
            <Icon iconName={iconName!} size="md" />
          </Pressable>
        )}
      </View>
      {Boolean(helperText) && <FieldHelperText text={helperText!} />}
      {/* {type === "numberRange" && value && (
        <NumberPickerActionSheet
          actionSheetRef={numberRangeActionSheetRef}
          max={max || (min || 0) + 100}
          min={min || 0}
          value={value}
          onChange={(result) => onChange(result)}
        />
      )}
      {type === "decimalRange" && value && (
        <DecimalRangeActionSheet
          actionSheetRef={decimalRangeActionSheetRef}
          max={max || (min || 0) + 100}
          min={min || 0}
          value={value}
          onChange={(result) => onChange(result)}
        />
      )} */}
      {/* {type === "height" && (
        <HeightActionSheet
          actionSheetRef={weightActionSheetRef}
          value={value}
          onChange={(result) => {
            onChange(result);
          }}
        />
      )} */}
    </View>
  );
};
