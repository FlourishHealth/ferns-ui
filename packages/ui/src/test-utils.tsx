import {render} from "@testing-library/react-native";
import React from "react";
import {ThemeProvider} from "./Theme";

export const renderWithTheme = (ui: React.ReactElement) => {
  return render(<ThemeProvider>{ui}</ThemeProvider>);
};

export const createCommonMocks = () => ({
  onChange: jest.fn(),
  onFocus: jest.fn(),
  onBlur: jest.fn(),
  onEnter: jest.fn(),
  onSubmitEditing: jest.fn(),
  onIconClick: jest.fn(),
});

export const setupComponentTest = () => {
  jest.useFakeTimers();
  return createCommonMocks();
};

export const teardownComponentTest = () => {
  jest.useRealTimers();
  jest.clearAllMocks();
};
