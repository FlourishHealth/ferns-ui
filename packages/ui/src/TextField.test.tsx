import {act, userEvent} from "@testing-library/react-native";
import React from "react";

import {renderWithTheme} from "./test-utils";
import {TextField} from "./TextField";

describe("TextField", () => {
  let mockOnChange: jest.Mock;
  let mockOnFocus: jest.Mock;
  let mockOnBlur: jest.Mock;
  let mockOnEnter: jest.Mock;

  beforeEach(() => {
    jest.useFakeTimers();
    mockOnChange = jest.fn();
    mockOnFocus = jest.fn();
    mockOnBlur = jest.fn();
    mockOnEnter = jest.fn();
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
  });

  describe("basic rendering", () => {
    it("should render with default props", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField value="test value" onChange={mockOnChange} />
      );

      expect(getByDisplayValue("test value").props.value).toBe("test value");
    });

    it("should render with title", () => {
      const {getByText} = renderWithTheme(
        <TextField title="Test Title" value="" onChange={mockOnChange} />
      );

      expect(getByText("Test Title")).toBeTruthy();
    });

    it("should render with placeholder", () => {
      const {getByPlaceholderText} = renderWithTheme(
        <TextField placeholder="Enter text" value="" onChange={mockOnChange} />
      );

      expect(getByPlaceholderText("Enter text")).toBeTruthy();
    });

    it("should render helper text", () => {
      const {getByText} = renderWithTheme(
        <TextField helperText="This is helper text" value="" onChange={mockOnChange} />
      );

      expect(getByText("This is helper text")).toBeTruthy();
    });

    it("should render error text", () => {
      const {getByText} = renderWithTheme(
        <TextField errorText="This is an error" value="" onChange={mockOnChange} />
      );

      expect(getByText("This is an error")).toBeTruthy();
    });
  });

  describe("user interactions", () => {
    it("should call onChange when text is entered", async () => {
      const user = userEvent.setup();
      const {getByDisplayValue} = renderWithTheme(<TextField value="" onChange={mockOnChange} />);

      const input = getByDisplayValue("");
      await user.type(input, "hello");

      expect(mockOnChange).toHaveBeenCalled();
      expect(mockOnChange.mock.calls.length).toBeGreaterThan(0);
    });

    it("should call onFocus when input is focused", async () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField value="" onChange={mockOnChange} onFocus={mockOnFocus} />
      );

      const input = getByDisplayValue("");
      expect(input.props.onFocus).toBeTruthy();
    });

    it("should call onBlur when input loses focus", async () => {
      const user = userEvent.setup();
      const {getByDisplayValue} = renderWithTheme(
        <TextField value="test" onBlur={mockOnBlur} onChange={mockOnChange} />
      );

      const input = getByDisplayValue("test");
      await user.press(input);
      await act(async () => {
        input.props.onBlur();
      });

      expect(mockOnBlur).toHaveBeenCalledTimes(1);
      expect(mockOnBlur.mock.calls[0][0]).toBe("test");
    });

    it("should call onEnter when enter key is pressed", async () => {
      const user = userEvent.setup();
      const {getByDisplayValue} = renderWithTheme(
        <TextField value="" onChange={mockOnChange} onEnter={mockOnEnter} />
      );

      const input = getByDisplayValue("");
      await act(async () => {
        input.props.onSubmitEditing();
      });

      expect(mockOnEnter).toHaveBeenCalledTimes(1);
    });
  });

  describe("field types", () => {
    it("should render email type with correct keyboard", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField type="email" value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input.props.keyboardType).toBe("email-address");
    });

    it("should render password type with secure text entry", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField type="password" value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input.props.secureTextEntry).toBe(true);
    });

    it("should render url type with correct keyboard", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField type="url" value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input.props.keyboardType === "url" || input.props.keyboardType === "default").toBe(
        true
      );
    });

    it("should render phoneNumber type with number keyboard", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField type="phoneNumber" value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input.props.keyboardType).toBe("number-pad");
    });

    it("should render search type with default keyboard", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField type="search" value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input.props.keyboardType).toBe("default");
    });
  });

  describe("multiline behavior", () => {
    it("should render as multiline when multiline prop is true", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField multiline value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input.props.multiline).toBe(true);
    });

    it("should set number of lines when rows prop is provided", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField multiline rows={5} value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input.props.numberOfLines).toBe(5);
    });

    it("should handle grow behavior with multiline", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField grow multiline value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input.props.multiline).toBe(true);
    });
  });

  describe("disabled state", () => {
    it("should be read-only when disabled", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField disabled value="test" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("test");
      expect(input.props.readOnly).toBe(true);
    });

    it("should not call onFocus when disabled", async () => {
      const user = userEvent.setup();
      const {getByDisplayValue} = renderWithTheme(
        <TextField disabled value="" onChange={mockOnChange} onFocus={mockOnFocus} />
      );

      const input = getByDisplayValue("");
      expect(input.props.readOnly).toBe(true);
    });

    it("should not call onBlur when disabled", async () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField disabled value="test" onBlur={mockOnBlur} onChange={mockOnChange} />
      );

      const input = getByDisplayValue("test");
      await act(async () => {
        input.props.onBlur();
      });

      expect(mockOnBlur).not.toHaveBeenCalled();
    });
  });

  describe("icon functionality", () => {
    it("should render icon when iconName is provided", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField iconName="check" value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input).toBeTruthy();
    });

    it("should call onIconClick when icon is pressed", async () => {
      const mockOnIconClick = jest.fn();
      const {getByDisplayValue} = renderWithTheme(
        <TextField
          iconName="check"
          value=""
          onChange={mockOnChange}
          onIconClick={mockOnIconClick}
        />
      );

      const input = getByDisplayValue("");
      expect(input).toBeTruthy();
    });
  });

  describe("accessibility", () => {
    it("should have correct accessibility properties", () => {
      const {getByDisplayValue} = renderWithTheme(<TextField value="" onChange={mockOnChange} />);

      const input = getByDisplayValue("");
      expect(input.props.accessibilityHint).toBe("Enter text here");
      expect(input.props["aria-label"]).toBe("Text input field");
    });

    it("should indicate disabled state in accessibility", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField disabled value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input.props.accessibilityState.disabled).toBe(true);
    });
  });

  describe("auto-complete and text content", () => {
    it("should set autoComplete property", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField autoComplete="username" value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input).toBeTruthy();
    });

    it("should handle text content type for email", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField type="email" value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input.props.textContentType).toBe("emailAddress");
    });

    it("should handle text content type for password", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField type="password" value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input.props.textContentType).toBe("password");
    });
  });

  describe("edge cases", () => {
    it("should handle empty value", () => {
      const {getByDisplayValue} = renderWithTheme(<TextField value="" onChange={mockOnChange} />);

      const input = getByDisplayValue("");
      expect(input.props.value).toBe("");
    });

    it("should handle undefined value", () => {
      const {root} = renderWithTheme(<TextField value={undefined} onChange={mockOnChange} />);

      expect(root).toBeTruthy();
    });

    it("should handle long text values", () => {
      const longText = "a".repeat(1000);
      const {getByDisplayValue} = renderWithTheme(
        <TextField value={longText} onChange={mockOnChange} />
      );

      const input = getByDisplayValue(longText);
      expect(input.props.value).toBe(longText);
    });

    it("should handle special characters", () => {
      const specialText = "!@#$%^&*()_+-=[]{}|;':\",./<>?";
      const {getByDisplayValue} = renderWithTheme(
        <TextField value={specialText} onChange={mockOnChange} />
      );

      const input = getByDisplayValue(specialText);
      expect(input.props.value).toBe(specialText);
    });
  });

  describe("return key behavior", () => {
    it("should set return key type", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField returnKeyType="done" value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input.props.enterKeyHint).toBe("done");
    });

    it("should handle blur on submit", () => {
      const {getByDisplayValue} = renderWithTheme(
        <TextField blurOnSubmit={false} value="" onChange={mockOnChange} />
      );

      const input = getByDisplayValue("");
      expect(input.props.blurOnSubmit).toBe(false);
    });
  });

  describe("input ref", () => {
    it("should call inputRef with the input reference", () => {
      const mockInputRef = jest.fn();
      renderWithTheme(<TextField inputRef={mockInputRef} value="" onChange={mockOnChange} />);

      expect(mockInputRef).toHaveBeenCalled();
    });
  });

  describe("snapshots", () => {
    it("should match snapshot with default props", () => {
      const component = renderWithTheme(<TextField value="test value" onChange={mockOnChange} />);
      expect(component.toJSON()).toMatchSnapshot();
    });

    it("should match snapshot with all props", () => {
      const component = renderWithTheme(
        <TextField
          disabled={false}
          errorText="Error text"
          helperText="Helper text"
          iconName="check"
          multiline={false}
          placeholder="Enter text"
          title="Test Title"
          type="text"
          value="test value"
          onBlur={mockOnBlur}
          onChange={mockOnChange}
          onEnter={mockOnEnter}
          onFocus={mockOnFocus}
          onIconClick={jest.fn()}
        />
      );
      expect(component.toJSON()).toMatchSnapshot();
    });

    it("should match snapshot when disabled", () => {
      const component = renderWithTheme(
        <TextField disabled title="Disabled Field" value="disabled value" onChange={mockOnChange} />
      );
      expect(component.toJSON()).toMatchSnapshot();
    });

    it("should match snapshot with multiline", () => {
      const component = renderWithTheme(
        <TextField
          multiline
          rows={3}
          title="Multiline Field"
          value="line 1\nline 2"
          onChange={mockOnChange}
        />
      );
      expect(component.toJSON()).toMatchSnapshot();
    });

    it("should match snapshot with error state", () => {
      const component = renderWithTheme(
        <TextField
          errorText="This field is required"
          title="Error Field"
          value=""
          onChange={mockOnChange}
        />
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
