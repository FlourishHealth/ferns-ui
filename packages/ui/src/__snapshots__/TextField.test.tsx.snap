// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField snapshots should match snapshot when disabled 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#1C1C1C",
        "fontSize": 14,
        "fontWeight": 600,
        "lineHeight": 22.4,
      }
    }
  >
    Disabled Field
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#D9D9D9",
        "borderColor": "#4E4E4E",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "overflow": "hidden",
        "paddingHorizontal": 12,
        "paddingVertical": 8,
      }
    }
  >
    <TextInput
      accessibilityHint="Enter text here"
      accessibilityState={
        {
          "disabled": true,
        }
      }
      aria-label="Text input field"
      autoCapitalize="sentences"
      autoCorrect={true}
      blurOnSubmit={true}
      keyboardType="default"
      numberOfLines={1}
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onContentSizeChange={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#686868"
      readOnly={true}
      secureTextEntry={false}
      style={
        {
          "color": "#1C1C1C",
          "flex": 1,
          "fontFamily": "text",
          "fontSize": 16,
          "gap": 10,
          "height": 20,
          "paddingVertical": 0,
          "width": "100%",
        }
      }
      textContentType="none"
      underlineColorAndroid="transparent"
      value="disabled value"
    />
  </View>
</View>
`;

exports[`TextField snapshots should match snapshot with all props 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#1C1C1C",
        "fontSize": 14,
        "fontWeight": 600,
        "lineHeight": 22.4,
      }
    }
  >
    Test Title
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginVertical": 2,
      }
    }
  >
    <View
      brand={false}
      color="#BD1111"
      duotone={false}
      light={false}
      name="triangle-exclamation"
      regular={false}
      sharp={false}
      size={12}
      solid={true}
      thin={false}
    />
    <View
      style={
        {
          "marginLeft": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#BD1111",
            "fontSize": 12,
          }
        }
      >
        Error text
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#D33232",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "overflow": "hidden",
        "paddingHorizontal": 12,
        "paddingVertical": 8,
      }
    }
  >
    <TextInput
      accessibilityHint="Enter text here"
      accessibilityState={
        {
          "disabled": false,
        }
      }
      aria-label="Text input field"
      autoCapitalize="sentences"
      autoCorrect={true}
      blurOnSubmit={true}
      keyboardType="default"
      multiline={false}
      numberOfLines={1}
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onContentSizeChange={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Enter text"
      placeholderTextColor="#686868"
      readOnly={false}
      secureTextEntry={false}
      style={
        {
          "color": "#1C1C1C",
          "flex": 1,
          "fontFamily": "text",
          "fontSize": 16,
          "gap": 10,
          "height": 20,
          "paddingVertical": 0,
          "width": "100%",
        }
      }
      textContentType="none"
      underlineColorAndroid="transparent"
      value="test value"
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      aria-role="button"
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        brand={false}
        color="#1C1C1C"
        duotone={false}
        light={false}
        name="check"
        regular={false}
        sharp={false}
        size={16}
        solid={true}
        thin={false}
      />
    </View>
  </View>
  <View
    style={
      {
        "marginTop": 2,
      }
    }
  >
    <Text
      style={
        {
          "color": "#1C1C1C",
          "fontSize": 12,
          "lineHeight": 16,
        }
      }
    >
      Helper text
    </Text>
  </View>
</View>
`;

exports[`TextField snapshots should match snapshot with default props 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#9A9A9A",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "overflow": "hidden",
        "paddingHorizontal": 12,
        "paddingVertical": 8,
      }
    }
  >
    <TextInput
      accessibilityHint="Enter text here"
      accessibilityState={
        {
          "disabled": undefined,
        }
      }
      aria-label="Text input field"
      autoCapitalize="sentences"
      autoCorrect={true}
      blurOnSubmit={true}
      keyboardType="default"
      numberOfLines={1}
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onContentSizeChange={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#686868"
      secureTextEntry={false}
      style={
        {
          "color": "#1C1C1C",
          "flex": 1,
          "fontFamily": "text",
          "fontSize": 16,
          "gap": 10,
          "height": 20,
          "paddingVertical": 0,
          "width": "100%",
        }
      }
      textContentType="none"
      underlineColorAndroid="transparent"
      value="test value"
    />
  </View>
</View>
`;

exports[`TextField snapshots should match snapshot with error state 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#1C1C1C",
        "fontSize": 14,
        "fontWeight": 600,
        "lineHeight": 22.4,
      }
    }
  >
    Error Field
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginVertical": 2,
      }
    }
  >
    <View
      brand={false}
      color="#BD1111"
      duotone={false}
      light={false}
      name="triangle-exclamation"
      regular={false}
      sharp={false}
      size={12}
      solid={true}
      thin={false}
    />
    <View
      style={
        {
          "marginLeft": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#BD1111",
            "fontSize": 12,
          }
        }
      >
        This field is required
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#D33232",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "overflow": "hidden",
        "paddingHorizontal": 12,
        "paddingVertical": 8,
      }
    }
  >
    <TextInput
      accessibilityHint="Enter text here"
      accessibilityState={
        {
          "disabled": undefined,
        }
      }
      aria-label="Text input field"
      autoCapitalize="sentences"
      autoCorrect={true}
      blurOnSubmit={true}
      keyboardType="default"
      numberOfLines={1}
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onContentSizeChange={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#686868"
      secureTextEntry={false}
      style={
        {
          "color": "#1C1C1C",
          "flex": 1,
          "fontFamily": "text",
          "fontSize": 16,
          "gap": 10,
          "height": 20,
          "paddingVertical": 0,
          "width": "100%",
        }
      }
      textContentType="none"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;

exports[`TextField snapshots should match snapshot with multiline 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#1C1C1C",
        "fontSize": 14,
        "fontWeight": 600,
        "lineHeight": 22.4,
      }
    }
  >
    Multiline Field
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#9A9A9A",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "overflow": "hidden",
        "paddingHorizontal": 12,
        "paddingVertical": 8,
      }
    }
  >
    <TextInput
      accessibilityHint="Enter text here"
      accessibilityState={
        {
          "disabled": undefined,
        }
      }
      aria-label="Text input field"
      autoCapitalize="sentences"
      autoCorrect={true}
      blurOnSubmit={true}
      keyboardType="default"
      multiline={true}
      numberOfLines={3}
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onContentSizeChange={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#686868"
      secureTextEntry={false}
      style={
        {
          "color": "#1C1C1C",
          "flex": 1,
          "fontFamily": "text",
          "fontSize": 16,
          "gap": 10,
          "height": 120,
          "paddingVertical": 0,
          "width": "100%",
        }
      }
      textContentType="none"
      underlineColorAndroid="transparent"
      value="line 1\\nline 2"
    />
  </View>
</View>
`;
